### ISSUE TITLE
Typescript Automatiion API uses incorrect argument (--remote-git-username)

### ISSUE THREAD

**Issue by alewis001** [pulumi/pulumi#12264]
### What happened?

Following the example here: https://github.com/pulumi/automation-api-examples/tree/main/nodejs/remoteDeployment-tsnode, and changing the `createOrSelectStack` to include Git username/password resulted in a failure `Error: unknown flag: --remote-git-username`

Note: Values in `<...>` are ones I've redacted.

Modified `createOrSelectStack`:
```typescript
const stack = await RemoteWorkspace.createOrSelectStack({
    stackName,
    url: "<repo_url>",
    branch: "refs/heads/main",
    projectPath: project,
    auth: {
      username: "x-token-auth",
      password: <repo_token>
    }
  });
```

Full Error:
```
CommandError: code: -2
 stdout:
 stderr: Command failed with exit code 1: pulumi up --yes --skip-preview --remote <repo_url> --remote-git-repo-dir <dir> --remote-git-branch refs/heads/main --remote-git-auth-password <repo_token> --remote-git-username x-token-auth --exec-kind auto.local --stack <stack> --non-interactive
Error: unknown flag: --remote-git-username
Usage:
  pulumi up [template|url] [flags]

...rest of pulumi command output
```

I can see in https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/localWorkspace.ts#L761 that it uses `--remote-git-username` instead of `--remote-git-auth-username`.

### Expected Behavior

The modified example completes without error.

### Steps to reproduce

1. Create new project using example: https://github.com/pulumi/automation-api-examples/tree/main/nodejs/remoteDeployment-tsnode
2. Modify `index.ts` to add `auth` section as per issue description, and to tailor according test environment and to provide a username and password to the auth section.
3. Run `npm run start <pulumi-username-or-organization>`

In my tests, I'm using Bitbucket which is why the username is `x-token-auth`. The password is an _Access Token_ created on the repository that holds my Pulumi program/stack.

### Output of `pulumi about`

Run within my automation project...
```
CLI
Version      3.55.0
Go Version   go1.20
Go Compiler  gc

Host
OS       darwin
Version  13.2.1
Arch     arm64

Backend
Name           pulumi.com
URL            https://app.pulumi.com/<username>
User           <username>
Organizations  <username>

Pulumi locates its logs in /var/folders/hf/8gb_vn0s55s3wk5j001prkq80000gp/T/ by default
warning: Failed to read project: no Pulumi.yaml project file found (searching upwards from <path>). If you have not created a project yet, use `pulumi new` to do so: no project file found
warning: Failed to get information about the current stack: no Pulumi.yaml project file found (searching upwards from <path>). If you have not created a project yet, use `pulumi new` to do so: no project file found
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
Thanks for opening the issue @alewis001, and sorry for the trouble. I've gone ahead and fixed this in #12269.

**Comment by alewis001** [comment]
Great, thanks @justinvp. No problem, just glad I found it and could raise it.
